tasks/jira.py

```diff
@@ -0,0 +1,127 @@
+import os
+import re
+import sys
+
+try:
+    from atlassian import Jira
+except ImportError:
+    pass
+from datadog_api_client import ApiClient, Configuration
+from datadog_api_client.v2.api.ci_visibility_tests_api import CIVisibilityTestsApi
+from datadog_api_client.v2.model.ci_app_aggregate_sort import CIAppAggregateSort
+from datadog_api_client.v2.model.ci_app_aggregation_function import CIAppAggregationFunction
+from datadog_api_client.v2.model.ci_app_compute import CIAppCompute
+from datadog_api_client.v2.model.ci_app_compute_type import CIAppComputeType
+from datadog_api_client.v2.model.ci_app_query_options import CIAppQueryOptions
+from datadog_api_client.v2.model.ci_app_sort_order import CIAppSortOrder
+from datadog_api_client.v2.model.ci_app_tests_aggregate_request import CIAppTestsAggregateRequest
+from datadog_api_client.v2.model.ci_app_tests_group_by import CIAppTestsGroupBy
+from datadog_api_client.v2.model.ci_app_tests_query_filter import CIAppTestsQueryFilter
+from invoke import task
+
+
+def get_failing_tests_names() -> set[str]:
+    """
+    Returns the names of the failing tests for the last 28 days
+    """
+
+    print('Getting failing tests for the last 28 days')
+    body = CIAppTestsAggregateRequest(
+        compute=[
+            CIAppCompute(
+                aggregation=CIAppAggregationFunction.COUNT,
+                metric="@test.full_name",
+                type=CIAppComputeType.TOTAL,
+            ),
+        ],
+        filter=CIAppTestsQueryFilter(
+            _from="now-28d",
+            query="@test.service:datadog-agent @git.branch:main @test.status:fail",
+            to="now",
+        ),
+        group_by=[
+            CIAppTestsGroupBy(
+                facet="@test.full_name",
+                limit=10000,
+                sort=CIAppAggregateSort(
+                    order=CIAppSortOrder.DESCENDING,
+                ),
+                total=False,
+            ),
+        ],
+        options=CIAppQueryOptions(
+            timezone="GMT",
+        ),
+    )
+
+    configuration = Configuration()
+    with ApiClient(configuration) as api_client:
+        api_instance = CIVisibilityTestsApi(api_client)
+        response = api_instance.aggregate_ci_app_test_events(body=body)
+        result = response['data']['buckets']
+        tests = {row['by']['@test.full_name'].removeprefix('github.com/DataDog/datadog-agent/') for row in result}
+
+        return tests
+
+
+def get_jira():
+    username = os.environ['ATLASSIAN_USERNAME']
+    password = os.environ['ATLASSIAN_PASSWORD']
+    jira = Jira(url="https://datadoghq.atlassian.net", username=username, password=password)
+
+    return jira
+
+
+def close_failing_test_issue(jira, issue_key: str, verbose_test: str):
+    print('Marking as done issue', issue_key, 'for test', verbose_test)
```

Comment:
Nit: should we use `won't do` instead?

Reply:
I think done is better here: we assume that if a test doesn't fail over 28 days, either it was an infra issue, or it has been fixed since while the card was kept open.

In both cases, we assume the task of fixing the specific failure is done. It would be hard to differentiate `WON'T DO` as they were an infra issue from `DONE` as it was fixed without closing the related card(s).